### PR TITLE
Replace CairoSVG with resvg-py and migrate PDF generation to reportlab

### PR DESCRIPTION
**Pull request by moshemorad**
## Summary
This PR replaces the LGPL-licensed CairoSVG library with the MIT-licensed resvg-py for SVG-to-PNG rasterization, and migrates PDF report generation from fpdf2 to reportlab. These changes improve licensing compliance and provide better control over report formatting.

### SVG Rasterization
- Replaced CairoSVG with resvg-py for converting SVG charts to PNG
- Updated `convert_svg_to_png()` in `core/reporting/utils.py` to use resvg-py instead of cairosvg
- Removed libcairo2 system dependency from Dockerfile
- Removed CairoSVG from playbooks/pyproject.toml

### PDF Report Generation
- Migrated `scanReportBlock_to_fileblock()` in `core/sinks/transformer.py` from fpdf2 to reportlab
- Replaced FPDF-based layout with reportlab's Platypus document model (SimpleDocTemplate, Table, Paragraph, Spacer)
- Implemented proper color handling using reportlab's Color objects with normalized RGB values
- Added `cell_markup()` helper to escape and convert markdown formatting (**bold** and newlines) to reportlab paragraph markup
- Improved table styling with TableStyle for borders, padding, and alignment
- Maintained landscape A4 page orientation and similar visual layout

### MS Teams Integration
- Updated `MsTeamsAdaptiveCardFilesImage` to use the new `convert_svg_to_png()` function

## Testing
- Added comprehensive test suite in `tests/test_scan_report_rendering.py` with:
  - PDF extraction utility to verify report contents without external PDF libraries
  - Tests for KRR and Popeye scan report generation
  - Tests for edge cases (empty results, hostile content with special characters)
- Added `tests/test_scan_sink_gating.py` to verify scan enrichments are properly converted before sink processing
- Added `tests/test_svg_conversion.py` to pin SVG-to-PNG conversion behavior
- Updated `tests/test_ai_integration.py` with PNG validation helpers and tests for graph tool output handling

## Implementation Details
- reportlab is imported lazily in the scan report path to avoid requiring the PDF toolchain for non-report use cases
- PDF text extraction in tests uses a custom implementation that decompresses content streams and extracts text without external PDF libraries
- The new implementation properly handles special characters, long text wrapping, and maintains consistent formatting across different scan types

https://claude.ai/code/session_01LTG88e6gtCnRrrf8Pgyhdo

### CODE REVIEW

File: src/robusta/core/sinks/transformer.py [hunk 1]
```diff
@@ -233,67 +231,94 @@ def scanReportBlock_to_fileblock(block: BaseBlock) -> BaseBlock:
...
-            rows = [scan_headers, *section_data]
-            write_section_header(pdf, section_name)
-            write_table(pdf, rows)
+        story = []
+
+        header_cells = [Paragraph(f"<b>{cell_markup(title)}</b> {scan.end_time.strftime('%b %d, %y %X')}", title_style)]
+        header_widths = [0.7]
+        if int(scan.score) >= 0:
+            header_cells.append(Paragraph(f"<b>{scan.grade}</b> {scan.score}", title_style))
+            header_widths.append(0.3)
```
[the remark this answers is not in the document]
Comment (moshemorad) [reply src/robusta/core/sinks/transformer.py]: Fixed in 2a0df9e, but with a `try/except` around the whole badge block instead of the suggested `float()` guard: `scan.grade` also parses the score internally via `int(self.score)`, so with the suggested fix a decimal score like `"85.5"` passes the `float()` check and then still raises `ValueError` on the `scan.grade` access one line later (verified by test). The committed version skips the badge with a warning for any unparseable score and is pinned by a parametrized test covering `"85.5"`, `"N/A"`, `""`, and `None`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pyproject.toml [hunk 2]
```diff
@@ -36,7 +36,9 @@ jinja2 = { version = ">=3.1.5", optional = true }
 grafana-api = { version = "^1.0.3", optional = true }
 watchdog =  { version = "^2.1.0", optional = true }
 better-exceptions = { version =  "^0.3.3", optional = true }
-CairoSVG = { version = "^2.5.2", optional = true }
+# SVG -> PNG rasterizer for charts sent to chat sinks (resvg is Apache-2.0 OR MIT,
+# bindings are MIT; replaced LGPL-licensed CairoSVG)
+resvg-py = "^0.3.4"
```
Comment (Avi-Robusta) [inline pyproject.toml]: not a well known package, would try to find alternatives with more users/stars
 if none are available pin exact version for security purposes

Comment (moshemorad) [reply pyproject.toml]: Surveyed the alternatives with more adoption — each was tested on a real robusta pygal chart, and none passes both the license and rendering requirements:

| Library | Downloads/mo | Why it doesn't work |
|---|---|---|
| PyMuPDF | 103.9M | AGPL-3.0 (stricter than the LGPL this PR removes) |
| cairosvg | 17.1M | LGPL-3.0 — the dependency being removed |
| svglib | ~2M | BSD, but crashes on our charts (`cairo.Error: invalid value for a dash setting`) and pulls the cairo stack back in |
| skia-python | 1.95M | BSD, but renders our charts as solid black — Skia's SVG module ignores `<style>` CSS blocks, which is where pygal puts all styling |
| Wand | 1.41M | ImageMagick delegates SVG to librsvg → LGPL native lib again |
| pyvips | 713K | libvips also rasterizes SVG via librsvg → LGPL |

resvg-py (882K/mo) is the most-adopted option that is both permissively licensed end-to-end and renders pygal output pixel-faithfully (validated side-by-side against cairosvg). The wrapper is thin; the engine is linebender/resvg, the standard Rust SVG renderer (~3.1K stars).

Since no qualifying alternative exists, pinned the exact version as suggested: `resvg-py = "0.3.4"` in 68c58e1 (lock file refreshed, no other dependency changes).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by Avi-Robusta — APPROVED** [review]
